language/move-lang/src/test_utils/mod.rs

```diff
@@ -20,6 +20,9 @@ pub const IR_EXTENSION: &str = "mvir";
 
 pub const COMPLETED_DIRECTORIES: &[&str; 2] = &["borrow_tests", "commands"];
 
+/// We need to replicate the specification of the (non-staged) stdlib files here since we can't
+/// import the stdlib crate: it will create a circular dependency since the stdlib needs the
+/// compiler to compile the stdlib and scripts.
```

Comment:
This unfortunately doesn't fully fix the problem. The functional tests will be run with the current/non-staged source files, but the data will be the staged genesis blob. So tests might type check, but they won't load or link